src/keyshare_export.rs

```diff
@@ -0,0 +1,162 @@
+//! Module to export and import key shares, with encryption in transit.
+
+use crate::{errors::Result, keygen::KeySharePrivate};
+use std::{marker::PhantomData, ops::Deref};
+
+mod pke;
+pub use pke::Pke;
+
+mod sodium;
+pub use sodium::SodiumPke;
+
+/// Struct for handling the export and import of KeySharePrivate
```

Comment:
I think we should simplify this type a little bit. I don't think the generic `T` here is really buying us much here. Can we remove this generic?

Reply:
Done.